### ISSUE TITLE
[trainer][metric] V1 computes response_length/clip_ratio against the padded batch width instead of the configured max length

### ISSUE THREAD

**Issue by YeonwooSung** [verl-project/verl#7817]
### System Info

Reproduced by reading current main (`1252cc71`).

Affects the default trainer (`trainer.use_v1: true` in `verl/trainer/config/ppo_trainer.yaml:228`):
- `verl/trainer/ppo/metric_utils.py` (`compute_data_metrics`, `_compute_response_info`)
- `verl/trainer/ppo/v1/trainer_base.py` (batch assembly, padding)
- `verl/trainer/ppo/v1/agent_loop_tq.py` (jagged prompt/response storage)

V0 is unaffected — `verl/experimental/agent_loop/agent_loop.py` pads to the configured length.

### Information

- [x] The official example scripts
- [ ] My own modified scripts

### Tasks

- [x] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [ ] My own task or dataset (give details below)

### Reproduction

`response_length/clip_ratio` and `prompt_length/clip_ratio` are meant to report the fraction of samples that hit the configured length cap, i.e. the truncation rate. They are computed by comparing each sample's length against a cap taken from the **tensor width**:

```python
# verl/trainer/ppo/metric_utils.py:459-460
max_prompt_length = batch.batch["prompts"].shape[-1]
max_response_length = batch.batch["responses"].shape[-1]
```

```python
# verl/trainer/ppo/metric_utils.py:576, 584, 592
"response_length/clip_ratio": torch.mean(torch.eq(response_length, max_response_length).float()).detach().item(),
"response_length_non_aborted/clip_ratio": non_aborted_response_length_clip_ratio,   # same max, line 517-519
"prompt_length/clip_ratio": torch.mean(torch.eq(prompt_length, max_prompt_length).float()).detach().item(),
```

That identity (`tensor width == configured cap`) holds on V0, where the agent loop pads every sample to the configured length before stacking:

```python
# verl/experimental/agent_loop/agent_loop.py:723-735
prompt_output = self._pad_token_ids(
    output.prompt_ids, max_length=self.rollout_config.prompt_length, padding_side="left", ...
)
response_output = self._pad_token_ids(
    output.response_ids, max_length=self.rollout_config.response_length, padding_side="right", ...
)
```

V1 does not. It stores raw jagged tensors (`verl/trainer/ppo/v1/agent_loop_tq.py:184-185`), then pads to the **batch maximum** while carrying the true per-sample lengths in separate columns:

```python
# verl/trainer/ppo/v1/trainer_base.py:1805-1835
prompt_length = data["prompts"].offsets().diff()
response_length = data["responses"].offsets().diff()
...
data = data.to_padded_tensor()          # width = longest sample in this batch
...
data["prompt_length"] = prompt_length.float()
data["response_length"] = response_length.float()
```

`_compute_response_info` then returns those true lengths:

```python
# verl/trainer/ppo/metric_utils.py:75-79
if "prompt_length" in batch.batch and "response_length" in batch.batch:
    return dict(
        prompt_length=batch.batch["prompt_length"],
        response_length=batch.batch["response_length"],
    )
```

So on V1 the comparison is *true length == longest length in this batch*, not *true length == configured cap*. Since some row always attains the batch maximum, the metric can never report 0, which is precisely the case users need it to report.

#### Concrete numbers

`data.max_response_length=4096`, `data.max_prompt_length=1024`, a batch of 4 trajectories with response lengths `[100, 200, 512, 512]` and prompt lengths `[50, 60, 70, 70]`. Nothing was truncated.

- `responses` is padded to width 512, `prompts` to width 70.
- Logged: `response_length/clip_ratio = mean([0,0,1,1]) = 0.5`, `prompt_length/clip_ratio = 0.5`, `response_length_non_aborted/clip_ratio = 0.5`.
- Correct: `0.0` and `0.0`.

Conversely, if every sample really does hit 4096, the width is 4096 and the ratio is 1.0 — correct by coincidence. The metric is wrong exactly in the regime where it is used to decide whether `max_response_length` needs raising.

`response_length/max` is still correct on V1 (it is a max over true lengths), so a quick way to see the inconsistency in an existing run's logs is that `response_length/clip_ratio` is nonzero while `response_length/max` is well below the configured `max_response_length`.

#### Minimal check

`compute_data_metrics` is a pure function, so this needs no GPU and no Ray — build a `DataProto` shaped the way V1 produces it (explicit `prompt_length` / `response_length` columns, `responses` width equal to the batch max rather than the configured cap) and assert the clip ratios are `0.0` when no sample reached the configured cap.

### Expected behavior

`response_length/clip_ratio` and `prompt_length/clip_ratio` should measure lengths against the configured caps (`data.max_response_length` / `data.max_prompt_length`, or the `actor_rollout_ref.rollout.response_length` / `.prompt_length` they feed), so the metric is 0 when nothing was truncated and identical between V0 and V1 for the same run.

The straightforward fix is to stop inferring the caps from tensor width and thread the configured values into `compute_data_metrics` — e.g. via `batch.meta_info` when V1 assembles the batch, falling back to `.shape[-1]` when they are absent so V0 and external callers keep working.

Happy to send a PR if that approach sounds right.
